c7n/resources/rdscluster.py

```diff
@@ -216,5 +211,141 @@ def process(self, snapshots):
+            elif port is None:
+                self.log_default_value('port')
+            else:
+                log.info('USING PARAMS PROVIDED IN POLICY')
+
+            print 'USING SNAPSHOT FOR CLUSTER::', latest_snapshot['DBClusterIdentifier']
+            print 'USING SNAPSHOT::', latest_snapshot['DBClusterSnapshotIdentifier']
+
+            params = {'DBClusterIdentifier': latest_snapshot['DBClusterIdentifier'],
+                      'SnapshotIdentifier': latest_snapshot['DBClusterSnapshotIdentifier'],
+                      'Engine': latest_snapshot['Engine'],
+                      'Port': port,
+                      'VpcSecurityGroupIds': vpc_sec_groups,
+                      'DBSubnetGroupName': db_subnet_group_name
+                      }
+
+            if snapshot_id:
+                params['SnapshotIdentifier'] = snapshot_id
+
+            try:
+                client.restore_db_cluster_from_snapshot(**params)
+                self.create_db_instances_within_cluster(latest_snapshot)
+            except ClientError as e:
+                raise
+
+    """
+      Right now im only returning the first value returned by the cli. I sort them from bottom to top then append the top
+      value to the array then return the array.  I am setting this function up to handle an array of different
+      cluster's snapshot, but for right now I just want to get the single cluster restore working then Ill focus on
+      creating multiple rds cluster instances that have same values within an array in the policy.
+    """
+    @staticmethod
+    def get_latest_snapshot(snapshots):
+
+        x = []
+        prev_obj = None
+
+        for snapshot in snapshots:
+
+            curr_obj = snapshot
+
+            if prev_obj is None:
+                prev_obj = curr_obj
+                x.append(curr_obj)
+
+            if curr_obj['DBClusterIdentifier'] == prev_obj['DBClusterIdentifier']:
+                prev_obj = curr_obj
+            else:
+                x.append(curr_obj)
+                prev_obj = curr_obj
+
+        print x
+
+        return x
+
+    def create_db_instances_within_cluster(self, snapshot):
+
+        db_instances = self.data.get('db-instances') or {}
+        num_of_instances = db_instances.get('number', 1)
+        db_identifier = db_instances.get('identifier', self.get_default_identifier(snapshot['DBClusterIdentifier']))
+        db_size = db_instances.get('size', 'db.r3.large')
+        post_fix = db_instances.get('post-fix', 'dev')
+
+        if db_identifier is None:
+            raise ClientError('MUST PROVIDE A db_identifier IN YOUR POLICY: Please delete cluster before trying again')
+        if db_size is None:
+            self.log_default_value('size:: default size is :: %s , which is minimum size', db_size)
+        if num_of_instances is None:
+            self.log_default_value('number:: number of db instances which is: %s', num_of_instances)
+
+        for instance in xrange(0, num_of_instances):
```

Comment:
s/xrange/range for any sizes where dealing with in resources, the difference is in consequential, and it removes a py3 incompatibilty.